consensus/hotstuff/persister/persister.go

```diff
@@ -1,69 +1,95 @@
 package persister
 
 import (
-	"github.com/dgraph-io/badger/v2"
+	"fmt"
 
 	"github.com/onflow/flow-go/consensus/hotstuff"
 	"github.com/onflow/flow-go/model/flow"
-	"github.com/onflow/flow-go/storage/badger/operation"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/operation"
 )
 
 // Persister is responsible for persisting minimal critical safety and liveness data for HotStuff:
 // specifically [hotstuff.LivenessData] and [hotstuff.SafetyData].
-//
-// Persister depends on protocol.State and cluster.State bootstrapping to set initial values for
-// SafetyData and LivenessData, for each distinct chain ID. This bootstrapping must be complete
-// before constructing a Persister instance with New (otherwise it will return an error).
 type Persister struct {
-	db      *badger.DB
+	db      storage.DB
 	chainID flow.ChainID
 }
 
 var _ hotstuff.Persister = (*Persister)(nil)
 var _ hotstuff.PersisterReader = (*Persister)(nil)
 
 // New creates a new Persister.
-//
-// Persister depends on protocol.State and cluster.State bootstrapping to set initial values for
-// SafetyData and LivenessData, for each distinct chain ID. This bootstrapping must be completed
-// before first using a Persister instance.
-func New(db *badger.DB, chainID flow.ChainID) (*Persister, error) {
+func New(db storage.DB, chainID flow.ChainID) (*Persister, error) {
+	err := ensureSafetyDataAndLivenessDataAreBootstrapped(db, chainID)
+	if err != nil {
+		return nil, fmt.Errorf("fail to check persister was properly bootstrapped: %w", err)
+	}
+
 	p := &Persister{
 		db:      db,
 		chainID: chainID,
 	}
 	return p, nil
 }
 
+// ensureSafetyDataAndLivenessDataAreBootstrapped checks if the safety and liveness data are
+// bootstrapped for the given chain ID. If not, it returns an error.
+// The Flow Protocol mandates that SafetyData and LivenessData is provided as part of the bootstrapping
+// data. For a node, the SafetyData and LivenessData is among the most safety-critical data. We require
+// the protocol.State's or cluster.State's bootstrapping logic to properly initialize these values in the
+// database.
+// It is highly recommended that components using the Persister read the current SafetyData
+// and/or LivenessData (depending on what they need) within their constructor as a sanity check.
```

Comment:
Now that `ensureSafetyDataAndLivenessDataAreBootstrapped` is called in the constructor, this documentation seems wrong. The Persister constructor is taking care of the the sanity check -- components using persister just need to check the error.